src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -631,6 +638,42 @@ public void addEnvVars(List<TemplateEnvVar> envVars) {
         }
     }
 
+    /**
+     * Deletes the template.
+     */
+    @POST
+    public HttpResponse doDoDelete(@AncestorInPath KubernetesCloud kubernetesCloud) throws IOException {
+        Jenkins j = Jenkins.get();
+        j.checkPermission(Jenkins.ADMINISTER);
+        if (kubernetesCloud == null) {
+            throw new IllegalStateException("Cloud could not be found");
+        }
+        kubernetesCloud.removeTemplate(this);
+        j.save();
+        // take the user back.
+        return new HttpRedirect("../../templates");
+    }
+
+    @POST
+    public HttpResponse doConfigSubmit(StaplerRequest req, @AncestorInPath KubernetesCloud kubernetesCloud) throws IOException, ServletException, Descriptor.FormException {
+        Jenkins j = Jenkins.get();
+        j.checkPermission(Jenkins.ADMINISTER);
+        if (kubernetesCloud == null) {
+            throw new IllegalStateException("Cloud could not be found");
+        }
+        kubernetesCloud.removeTemplate(this);
+        PodTemplate newTemplate = reconfigure(req, req.getSubmittedForm());
+        kubernetesCloud.addTemplate(newTemplate);
+        j.save();
+        // take the user back.
+        return FormApply.success("../../templates");
+    }
+
+    private PodTemplate reconfigure(@NonNull final StaplerRequest req, JSONObject form) throws Descriptor.FormException {
+        if (form == null)     return null;
```

Comment:
```suggestion
        if (form == null) {
            return null;
        }
```

For code style consistency.